ddtrace/opentracer/tracer.go

```diff
@@ -84,3 +87,12 @@ func (t *opentracer) Extract(format interface{}, carrier interface{}) (opentraci
 		return nil, opentracing.ErrUnsupportedFormat
 	}
 }
+
+// ContextWithSpan implements opentracing.TracerContextWithSpanExtension.
+func (t *opentracer) ContextWithSpanHook(ctx context.Context, openSpan opentracing.Span) context.Context {
+	ddSpan, ok := openSpan.(*span)
+	if !ok {
+		return ctx
+	}
+	return tracer.ContextWithSpan(ctx, ddSpan.Span)
```

Comment:
One thing though. Is it worth or possible to add a test for this?

Reply:
test added